utils/registry/registry_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2018 Red Hat, Inc.
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package core manages the lifecycle of all plugins (start, graceful
+// shutdown) and defines the core lifecycle SPI. The core lifecycle SPI
+// must be implemented by each plugin.
+
+package registry_test
```

Comment:
Curious why package names are different? if resgistry_test.go tests registry.go functionality, I think same package name should be used to have complete access to vars defined in registry package.

Reply:
@sbezverk Using a different package name for tests is considered to be best practice because it means your tests do not have access to package private info.

If I used package registry to test package registry, some tests may pass because they are in package registry, even though functionality *does* break from outside the package.

So you use ${package}_test instead.  Go implicitly encourages it because you will get an error mixing two packages in the same subdir *unless* one of ${package}_test.